bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/handler/GardenaAccountHandler.java

```diff
@@ -51,80 +64,188 @@
+    }
+
+    /**
+     * Get the duration remaining until the end of the api call suppression window, or Duration.ZERO if we are outside
+     * the call suppression window.
+     *
+     * @return the duration until the end of the suppression window, or zero.
+     */
+    private Duration getApiCallSuppressionDurationRemaining() {
+        Instant now = Instant.now();
+        if (now.isAfter(apiCallSuppressionStart) && now.isBefore(apiCallSuppressionEnd)) {
+            return Duration.between(now, apiCallSuppressionEnd);
+        }
+        return Duration.ZERO;
+    }
+
+    /**
+     * Updates the time when api call suppression begins to now(), and the time when api call suppression ends to now()
+     * plus the given delay. The delay must not be a negative duration. Saves the start and end time values as
+     * properties to ensure consistent behaviour across restarts.
+     *
+     * @param delay the duration between the start and end of the suppression window.
+     */
+    private void apiCallSuppressionPropertiesUpdate(Duration delay) {
+        Duration offset = delay.isNegative() ? Duration.ZERO : delay;
+        apiCallSuppressionStart = Instant.now();
+        apiCallSuppressionEnd = apiCallSuppressionStart.plus(offset);
+        Thing thing = getThing();
+        thing.setProperty(GardenaBindingConstants.API_CALL_SUPPRESSION_START, apiCallSuppressionStart.toString());
+        thing.setProperty(GardenaBindingConstants.API_CALL_SUPPRESSION_END, apiCallSuppressionEnd.toString());
     }
 
     @Override
     public void initialize() {
         logger.debug("Initializing Gardena account '{}'", getThing().getUID().getId());
-        initializeGardena();
+        loadApiCallSuppressionProperties();
+        Duration delay = getApiCallSuppressionDurationRemaining();
+        if (Duration.ZERO.equals(delay)) {
+            // do immediate initialisation
+            scheduler.submit(() -> initializeGardena());
+        } else {
+            // delay the initialisation
+            scheduleReinitialize(delay);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, getUiText());
+        }
     }
 
     public void setDiscoveryService(GardenaDeviceDiscoveryService discoveryService) {
         this.discoveryService = discoveryService;
     }
 
+    /**
+     * Format a localised description text to display on the thing main UI page.
+     *
+     * @return the (localised) description text.
+     */
+    private String getUiText() {
+        String template = i18nProvider.getText(bundle, RECONNECT_MSG_KEY, RECONNECT_MSG_KEY,
+                localeProvider.getLocale());
+        String dateTime = LocalDateTime.ofInstant(apiCallSuppressionEnd, timeZoneProvider.getTimeZone())
+                .truncatedTo(ChronoUnit.SECONDS).format(DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+        return MessageFormat.format(template, dateTime);
```

Comment:
Why don't you simply pass `dateTime` as last parameter to `getText`? The additional `format()` call is not needed.

Reply:
I tried to do that, but I couldn't figure out a way around the compiler warning _"Null type mismatch (type annotations): required '@NonNull String' but this expression has type '@Nullable String'"_

![image](https://user-images.githubusercontent.com/893994/181185602-918efa82-1b33-4a6e-aaf1-acc90374f642.png)

Reply:
So I suppose the correct syntax is as follows ... yuk!

```
private String getUiText() {
    String msg = i18nProvider.getText(bundle, RECONNECT_MSG_KEY, RECONNECT_MSG_KEY, localeProvider.getLocale(),
            LocalDateTime.ofInstant(apiCallSuppressionEnd, timeZoneProvider.getTimeZone())
                    .truncatedTo(ChronoUnit.SECONDS).format(DateTimeFormatter.ISO_LOCAL_DATE_TIME));
    return msg != null ? msg : "";
}
```

Reply:
I think the most correct fix is making the return value nullable. Suggestion (including some renaming suggestions):

```
private @Nullable String getReconnectUiMessage() {
    LocalDateTime timestamp = LocalDateTime.ofInstant(apiCallSuppressionEnd, timeZoneProvider.getTimeZone())
            .truncatedTo(ChronoUnit.SECONDS);
    DateTimeFormatter formatter = [...]; // see below comment
    return i18nProvider.getText(bundle, RECONNECT_MSG_KEY, null, localeProvider.getLocale(), timestamp.fornat(formatter));
}
```

That way, the message will simply be omitted in the (impossible) case of the string resource not being found.

Reply:
Ok. Done.